django/db/models/expressions.py

```diff
@@ -474,10 +457,27 @@ def as_sql(self, compiler, connection):
         return expression_wrapper % sql, expression_params
 
     def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize=False, for_save=False):
+        lhs = self.lhs.resolve_expression(query, allow_joins, reuse, summarize, for_save)
+        rhs = self.rhs.resolve_expression(query, allow_joins, reuse, summarize, for_save)
+        if self.__class__ is CombinedExpression:
+            try:
+                lhs_type = lhs.output_field.get_internal_type()
+            except (AttributeError, FieldError):
```

Comment:
`AttributeError` is raised, when source expression is resolved to either `OuterRef` or `ResolvedOuterRef`.